src/services/IPRService.ts

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-console */
+import { Logger } from "@aws-lambda-powertools/logger";
+import { AppError } from "../utils/AppError";
+import { DynamoDBDocument, GetCommand, UpdateCommand } from "@aws-sdk/lib-dynamodb";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { IPREventTypeToAttributeMapper } from "./IPREventTypeToAttributeMapper";
+import { Constants } from "../utils/Constants";
+
+export class IPRService {
+	readonly tableName: string;
+
+	private readonly dynamo: DynamoDBDocument;
+
+	readonly logger: Logger;
+
+	private static instance: IPRService;
+
+	private readonly iprEventTypeToAttributeMapper: IPREventTypeToAttributeMapper;
+
+	constructor(tableName: any, logger: Logger, dynamoDbClient: DynamoDBDocument) {
+		this.tableName = tableName;
+		this.dynamo = dynamoDbClient;
+		this.logger = logger;
+		this.iprEventTypeToAttributeMapper = new IPREventTypeToAttributeMapper();
+	}
+
+	static getInstance(tableName: string, logger: Logger, dynamoDbClient: DynamoDBDocument): IPRService {
+		if (!IPRService.instance) {
+			IPRService.instance = new IPRService(tableName, logger, dynamoDbClient);
+		}
+		return IPRService.instance;
+	}
+
+	async isFlaggedForDeletionOrEventAlreadyProcessed(userId: string, eventType: string): Promise<boolean | undefined> {
+		this.logger.info({ message: "Checking if record is flagged for deletion or already processed", tableName: this.tableName, userId });
+		const getSessionCommand = new GetCommand({
+			TableName: this.tableName,
+			Key: {
+				userId,
+			},
+		});
+
+		try {
+			const session = await this.dynamo.send(getSessionCommand);
+			const eventAttribute = this.iprEventTypeToAttributeMapper.map(eventType);
```

Comment:
this method could probably be inlined, or moved to this class, to improve understanding of the processing happening here.